daemon/graphdriver/aufs/aufs.go

```diff
@@ -308,11 +308,17 @@ func (a *Driver) Put(id string) error {
 
 // Diff produces an archive of the changes between the specified
 // layer and its parent layer which may be "".
-func (a *Driver) Diff(id, parent string) (archive.Archive, error) {
+func (a *Driver) Diff(id, parent string, excludes []string) (archive.Archive, error) {
+	// archive.TarWithOptions uses relative path to compare excludes.
+	// So we have to strip the leading '/'.
+	newExcludes := make([]string, len(excludes))
+	for i, ex := range excludes {
+		newExcludes[i] = strings.TrimLeft(ex, "/")
```

Comment:
Do you ever actually check to make sure the incoming paths are absolute?  I didn't notice it. I think we should generate an error if they're not.

Not 100% sure of the best place to do the check but this func seems reasonable since you want to catch all code paths that might hit this func.  Of course, putting the check in the archive code might not be bad either, since if you're correct that its assuming relative paths then it really should enforce it.